tests/test_mutations.py

```diff
@@ -1022,6 +1024,63 @@ def verify_model_ts_general(self, ts, model, discrete, verify_roots, mutation_ra
             expected_roots = expected_ancestral_states * ts.num_sites
             self.chisquare(roots, expected_roots)
 
+    def verify_model_ts_general(self, ts, model, discrete, verify_roots, mutation_rate):
+        alleles = [a.decode() for a in model.alleles]
+        num_alleles = len(alleles)
+        roots = np.zeros((num_alleles,))
+
+        observed_transitions = np.zeros((num_alleles, num_alleles))
+        for mut in ts.mutations():
+            if mut.parent == tskit.NULL:
+                pa = ts.site(mut.site).ancestral_state
+            else:
+                pa = ts.mutation(mut.parent).derived_state
+            da = mut.derived_state
+            observed_transitions[alleles.index(pa), alleles.index(da)] += 1
+        pth = 0.05 / num_alleles
+        for j, (row, p) in enumerate(
+            zip(observed_transitions, model.transition_matrix)
+        ):
+            p[j] = 0
+            p /= sum(p)
+            self.chisquare(row, sum(row) * p, pth)
+
+        if verify_roots:
+            for site in ts.sites():
+                aa = site.ancestral_state
+                roots[alleles.index(aa)] += 1
+
+            expected_ancestral_states = np.zeros(num_alleles)
+            change_probs = model.transition_matrix.sum(axis=1) - np.diag(
+                model.transition_matrix
+            )
+            for t in ts.trees():
+                if discrete:
+                    t_span = np.ceil(t.interval[1] - np.ceil(t.interval[0]))
+                    expected_ancestral_states += (
+                        model.root_distribution
+                        * t_span
+                        * (
+                            1
+                            - np.exp(
+                                -mutation_rate * t.total_branch_length * change_probs
+                            )
+                        )
+                    )
+                else:
+                    t_span = t.span
+                    expected_ancestral_states += (
+                        model.root_distribution
+                        * mutation_rate
+                        * t.total_branch_length
+                        * t.span
+                        * change_probs
+                    )
+
+            expected_ancestral_states /= expected_ancestral_states.sum()
+            expected_roots = expected_ancestral_states * ts.num_sites
+            self.chisquare(roots, expected_roots)
```

Comment:
I think that you should just do `self.chisquare(roots, expected_ancestral_states)`, without renormalizing it here?